documents/payment.md

```diff
@@ -466,6 +466,191 @@ body, err := client.Payment.CreatePaymentJson(para_attr, nil)
+
+**Response:** <br>
+```json
+{
+  "razorpay_payment_id": "pay_EAm09NKReXi2e0"
+}
+```
+-------------------------------------------------------------------------------------------------------
+### Create Payment UPI s2s / VPA token (Third party validation)
+
+```go
+para_attr := map[string]interface{}{
+    "amount": 100,
+    "currency": "INR",
+    "order_id": "order_Ee0biRtLOqzRjP",
+    "email": "gaurav.kumar@example.com",
+    "contact": "9090909090",
+    "method": "upi",
+    "ip": "192.168.0.103",
+    "referer": "http",
+    "user_agent": "Mozilla/5.0",
+    "description": "Test flow",
+    "notes": map[string]interface{}{
+      "purpose": "UPI test payment",
+    },
+    "upi": map[string]interface{}{
+      "flow" : "intent",
+    },
+}
+body, err := client.Payment.CreateUpi(para_attr, nil)
+```
+
+**Parameters:**
+| Name        | Type    | Description                          |
+|-------------|---------|--------------------------------------|
+| amount*          | integer | Amount of the order to be paid  |
+| currency*   | string  | The currency of the payment (defaults to INR)                                  |
+| order_id*        | string  | The unique identifier of the order created. |
+| email*        | string      | Email of the customer                       |
+| customer_id*   | string      | The id of the customer to be fetched |
+| contact*      | string      | Contact number of the customer              |
+| notes | array  | A key-value pair  |
+| description | string  | Descriptive text of the payment. |
+| save | boolean  |  Specifies if the VPA should be stored as tokens.Possible value is `0`, `1`  |
+| callback_url   | string      | URL where Razorpay will submit the final payment status. |
+| ip*   | string      | The client's browser IP address. For example `117.217.74.98` |
+| referer*   | string      | Value of `referer` header passed by the client's browser. For example, `https://example.com/` |
+| user_agent*   | string      | Value of `user_agent` header passed by the client's browser. For example, `Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36` |
+| upi* (for Upi only)  | array      | All keys listed [here](https://razorpay.com/docs/payments/third-party-validation/s2s-integration/upi/intent/#step-2-initiate-a-payment) are supported  |
+
+**Response:** <br>
+```json
+{
+  "razorpay_payment_id": "pay_CMeM6XvOPGFiF",
+  "link": "upi://pay?pa=success@razorpay&pn=xyz&tr=xxxxxxxxxxx&tn=gourav&am=1&cu=INR&mc=xyzw"
+}
+```
+-------------------------------------------------------------------------------------------------------
+
+### Valid VPA (Third party validation)
+
+```go
+para_attr := map[string]interface{}{
+  "vpa": "gauravkumar@exampleupi",
+}
+
+body, err := client.Payment.ValidateVpa(para_attr, nil)
```

Comment:
client.Payment.ValidateVpa undefined

Reply:
try with this 
`para_attr := map[string]interface{}{
     "vpa": "success@razorpay",
 }`
	  
`body, err := client.Payment.ValidateVpa(para_attr, nil)`